pkg/services/authz/zanzana/server/server_list.go

```diff
@@ -0,0 +1,143 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"github.com/grafana/grafana/pkg/apimachinery/utils"
+	authzextv1 "github.com/grafana/grafana/pkg/services/authz/zanzana/proto/v1"
+)
+
+func (s *Server) List(ctx context.Context, r *authzextv1.ListRequest) (*authzextv1.ListResponse, error) {
+	ctx, span := tracer.Start(ctx, "authzServer.List")
+	defer span.End()
+
+	if info, ok := typeInfo(r.GetGroup(), r.GetResource()); ok {
+		return s.listTyped(ctx, r, info)
+	}
+
+	return s.listGeneric(ctx, r)
+}
+func (s *Server) listTyped(ctx context.Context, r *authzextv1.ListRequest, info TypeInfo) (*authzextv1.ListResponse, error) {
+	// 1. check if subject has access through namespace because then they can read all of them
+	res, err := s.openfga.Check(ctx, &openfgav1.CheckRequest{
+		StoreId:              s.storeID,
+		AuthorizationModelId: s.modelID,
+		TupleKey: &openfgav1.CheckRequestTupleKey{
+			User:     r.GetSubject(),
+			Relation: mapping[utils.VerbGet],
```

Comment:
Could we make it more flexible and possible to use other verbs here?

Reply:
Added in https://github.com/grafana/grafana/pull/95056/commits/60e399744de8d421d32c2a36779d1f03e48bbeea